### DISCUSSION TITLE
lldp neighbor interface ips getting swapped.

### DISCUSSION THREAD

**Discussion by SandipJunghare** [lldpd/lldpd#505] [Q&A]
Hello Vincent, 

We are working on LLDP, and we came across your source([https://github.com/lldpd/lldpd](https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flldpd%2Flldpd&data=04%7C01%7CSandip.Junghare%40einfochips.com%7C06dc7c6a8e084cbf733108da08140e48%7C0beb0c359cbb4feb99e5589e415c7944%7C1%7C0%7C637831179510113773%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=nCUZlfMib%2BtZsnBFSzPy6ADV%2FW8wmsUfNhklKSDFC4Q%3D&reserved=0)) which we started using on our hardware. The hardware boards have 2 ethernet ports namely eth0 and eth1. For our test setup, we connected our 2 hardware boards through eth0 ports (physical connection). 
We have assigned static IPs on eth1 ports of both the boards and IP addresses through avahi on eth0 ports on the 2 hardwares respectively. The following image shows the IPs assigned on both boards.
![image](https://user-images.githubusercontent.com/88190551/158817678-8b78fede-3ff2-4da2-8141-486b2ff95c86.png) 
![image](https://user-images.githubusercontent.com/88190551/158817721-0cbe185d-4579-4490-ae8f-df97794a66bf.png)

                         
We are trying to get the neighbour's information on the respective boards using lldp(your source).  We are facing issues as we are expected to view the eth0 information but eth1 information is available on their respective board's consoles.

![image](https://user-images.githubusercontent.com/88190551/158817793-cca59cbb-2feb-49b4-b2ac-6db809fa290a.png) 
![image](https://user-images.githubusercontent.com/88190551/158817820-50963291-7201-41ff-9140-064768f9acbf.png)

              
Could you let us know if we are missing out something?

Regards,
Sandip.

**Comment by vincentbernat** [comment]
The standard defines the management IP as a global information attached to the agent. Therefore, it is not per-port. By default, lldpd will use the first available IP address (in the interface order). Once it has chosen an IP address, it will not change it (this may be a problem when using DHCP). Look at the `-m` flag to change the chosen IP address or at the `configure system ip management pattern` command.

**Reply by TripuraPasnur11** [reply]
Hello Vincent,
I have tested with the _**configure system IP management pattern**_ command by changing the management IP in the configuration and I am able to view the neighbor's information(i.e. eth0:avahi).  
The below image shows the interfaces on the device:
![image](https://user-images.githubusercontent.com/102207920/159633547-dd9f69d1-893a-47bb-a9df-9575a25b0aaf.png)
I have changed the interface pattern in the configuration file by using **_configure system interface pattern_** (check configuration image below) and I want to view the neighbor's information(eth0:avahi) through this. 
![image](https://user-images.githubusercontent.com/102207920/159634023-54609f25-2ee7-4ace-8841-1ccf64f33f98.png)

But when I try to test, I am unable to view the neighbor's information(eth0:avahi port's info). Can you guide what else needs to be done by just mentioning the interface in the configuration without a need to add a management IP address in the configuration? I  referred to this link: https://lldpd.github.io/usage.html 

Thanks,
Tripura

**Reply by vincentbernat** [reply]
`eth0:avahi` is not an interface. `ifconfig` output is misleading. Look at `ip addr show` instead. The interface is `eth0`.
